Support for alternative container runtimes - Podman (#2745)
dapr/docs: daprdocs/content/en/reference/cli/dapr-uninstall.md

| `--help`, `-h`       |                      |               | Print this help message                                                                                                                             |
| `--kubernetes`, `-k` |                      | `false`       | Uninstall Dapr from a Kubernetes cluster                                                                                                            |
| `--namespace`, `-n`  |                      | `dapr-system` | The Kubernetes namespace to uninstall Dapr from                                                                                                     |

> |  `--container-runtime`  |              |    docker      | It is used to pass in a different container runtime other than docker. Supported container runtime are docker/podman|

Reviewer: ```suggestion
|  `--container-runtime`  |              |    `docker`      | Used to pass in a different container runtime other than Docker. Supported container runtimes are: `docker`, `podman` |
```

dapr uninstall --all
```

> You can specify a different container runtime while uninstalling dapr. If you omit this flag then default container runtime is docker.

Reviewer: ```suggestion
You can specify a different container runtime while setting up Dapr. If you omit the `--container-runtime` flag, the default container runtime is Docker.
```

> You can specify a different container runtime while setting up Dapr. If you omit the `--container-runtime` flag, the default container runtime is Docker.

Reviewer 2: If I try to uninstall and I do not provide podman, does this fail gracefully and give me an understandable error?

```suggestion
You can specify a different container runtime uninstalling Dapr. If you omit the `--container-runtime` flag, the default container runtime is Docker.
```

Reviewer 2: @pravinpushkar - Re this. https://github.com/dapr/cli/pull/1041#issuecomment-1255004312 Does this work on LInux?

Reviewer 2: It would still be beneficial to Create a "Run with podman" topic and copy the init, run and uninstall sections from here  https://docs.dapr.io/operations/hosting/self-hosted/self-hosted-with-docker/ and include this as a separate heading in the index. That way it does make it show that Dapr is not dominated by Docker and shows how to use podman more detailed steps on how to use podman, including setting this up.

Author: @msfussell Podman can be installed on all platforms- https://podman.io/getting-started/installation.html
So, I think there should not be any issue with dapr using Podman. Currently our GH workflow for macos uses Podman.

Author: @msfussell Added a separate page for podman just like https://docs.dapr.io/operations/hosting/self-hosted/self-hosted-with-docker/. Podman commands are similar to Docker and dapr run should behave in the same manner. I have the content for what I was able to verify.
